Common/Algorithm/Framework/Alphas/Analysis/Providers/AlgorithmSecurityValuesProvider.cs

```diff
@@ -55,15 +54,14 @@ public SecurityValues GetValues(Symbol symbol)
         /// <returns>The insight target values for all the algorithm securities</returns>
         public ReadOnlySecurityValuesCollection GetAllValues()
         {
-            var values = new Dictionary<Symbol, SecurityValues>();
-            foreach (var kvp in _algorithm.Securities)
-            {
-                var security = kvp.Value;
-                var volume = security.Cache.GetData<TradeBar>()?.Volume ?? 0;
-                var value = new SecurityValues(security.Symbol, _algorithm.UtcTime, security.Exchange.Hours, security.Price, security.VolatilityModel.Volatility, volume, security.QuoteCurrency.ConversionRate);
-                values[security.Symbol] = value;
-            }
-            return new ReadOnlySecurityValuesCollection(values);
+            // lets be lazy creating the SecurityValues
+            return new ReadOnlySecurityValuesCollection(
```

Comment:
This likely saves a ton of time, just curious if any negative impact was observed when dealing with alphas with a large number of insights?